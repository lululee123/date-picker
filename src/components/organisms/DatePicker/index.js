import React from 'react';
import useOnclickOutside from 'react-cool-onclickoutside';
import classnames from 'classnames';

import { useSelector, useDispatch } from 'react-redux';

import { openStatus, dateChange } from '../../../model/calendar';

import Calendar from '../Calendar';
import DatePickerInput from '../DatePickerInput';

import styles from './index.scss';

const DatePicker = ({ date = '', onSelect = () => {}, className }) => {
	const { calendarInfo: { open } } = useSelector(state => state.calendar);
	const dispatch = useDispatch();

	const ref = useOnclickOutside(() => {
		dispatch(openStatus(false));
	});

	const handleClickBtn = () => {
		dispatch(openStatus(!open));
	};

	const calendarClick = selectDate => {
		dispatch(dateChange(selectDate));
		onSelect();
	};

	return (
		<div ref={ref} className={classnames(styles.datePicker, className)}>
			<DatePickerInput date={date} open={open} onClick={handleClickBtn} />
			{open && <Calendar date={date} onSelect={calendarClick} />}
		</div>
	);
};

export default DatePicker;
