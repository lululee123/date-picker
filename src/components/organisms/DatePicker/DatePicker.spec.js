import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../../../store';

import DatePicker from './index';

test('render DatePicker correctly', () => {
	const component = renderer
		.create(
			<Provider store={store}>
				<DatePicker />
			</Provider>,
		)
		.toJSON();
	expect(component).toMatchSnapshot();
});
