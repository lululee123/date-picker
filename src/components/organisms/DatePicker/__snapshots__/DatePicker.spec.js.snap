// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DatePicker correctly 1`] = `
<div
  className="datePicker"
>
  <span
    className="datePickerInput"
  >
    <button
      className="icon calendarIcon"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="#666"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 16 16"
        width="100%"
      >
        <path
          d="M15 1h-2v-1h-3v1h-4v-1h-3v1h-2c-0.552 0-1 0.448-1 1v0 13c0 0.552 0.448 1 1 1v0h14c0.552 0 1-0.448 1-1v0-13c0-0.552-0.448-1-1-1v0zM15 4v3h-4v-3h4zM11 1h1v2h-1v-2zM10 4v3h-4v-3h4zM10 8v3h-4v-3h4zM4 1h1v2h-1v-2zM5 4v3h-4v-3h4zM1 8h4v3h-4v-3zM1 15v-3h4v3h-4zM6 15v-3h4v3h-4zM15 15h-4v-3h4v3zM11 11v-3h4v3h-4z"
        />
      </svg>
    </button>
    <input
      className="calendarInput"
      onClick={[Function]}
      readOnly="readonly"
      value=""
    />
  </span>
</div>
`;
